### PR TITLE
Added docgen info to Vite output for SFC components

### LINKED ISSUE #46: Controls not automatically generated for Single File Components
Hi

Thanks so much for writing this builder for Storybook! Being able to use Vite is so much better!

I am having an issue with getting controls to automatically generate for single file components though.

I have followed the setup guide for getting started using Storybook with Vue 3, Vite and TypeScript. Everything is running ok except I cannot get my story to automatically generate controls (or source) based on the components props.

I've had controls configured using the `argTypes` property but as far as I am aware Storybook should be able to generate these out of the box based on the components props?

I have been hunting through documentation and addons but can't find anything. Any help would be much appreciated.

Here is my current setup:

**Storybook Packages**:
- "@babel/core": "^7.14.6"
- "@storybook/addon-actions": "^6.3.0"
- "@storybook/addon-essentials": "^6.3.0"
- "@storybook/addon-links": "^6.3.0"
- "@storybook/vue3": "^6.3.0"
- "babel-loader": "^8.2.2"
- "storybook-builder-vite": "^0.0.9"
- "typescript": "^4.3.2"

**main.js**:
```javascript
module.exports = {
    "stories": [
        "../src/**/*.stories.mdx",
        "../src/**/*.stories.@(js|jsx|ts|tsx)"
    ],
    "addons": [
        "@storybook/addon-links",
        "@storybook/addon-essentials"
    ],
    "core": {
        "builder": "storybook-builder-vite"
    }
};
```

**preview.js**:
```javascript
export const parameters = {
    actions: {
        argTypesRegex: "^on[A-Z].*"
    },
    controls: {
        expanded: true,
        matchers: {
            color: /(background|color)$/i,
            date: /Date$/,
        },
    },
};
```

**The source component props**:
```typescript
props: {

    /**
        * The unique id of this dropdown panel
    */
    id: {
        type: String,
        default: () => stringUniqueId(6),
    },

    /**
        * Where the dropdown panel originates from.
        * This could also be described as it's anchor point.
    */
    origin: {
        type: String as PropType<Origin>,
        default: 'bottom-centre',
    },

    /**
    * A boolean indicating whether this dropdown panel should be teleported
    */
    teleport: {
        type: Boolean,
        default: false,
    },

    /**
    * A boolean indicating whether the panel should automatically
    * close when an item within the panel is clicked.
    */
    autoClose: {
        type: Boolean,
        default: true,
    },

    /**
        * A value indicating how autofocus should behave within the panel
    */
    autoFocus: {
        type: [Boolean, String],
        default: 'input[type="text"]|[active]|:focusable',
    },

    /**
        * A value indicating whether the dropdown panel should automatically trap focus upon opening
    */
    focusTrap: {
        type: Boolean,
        default: true,
    },

},
```

**The story for this component**:
```typescript
import DropdownPanel from '../../components/dropdown-panel/dropdown-panel.vue';

import template from './templates/default.html?raw';
import hero from '../_base/decorators/hero';

import {
    Meta,
    Story,
} from '@storybook/vue3';

export default {
    title: 'Dropdown Panel',
    component: DropdownPanel,
    decorators: [ hero ],
} as Meta;

const Component: Story = args => ({

    template,

    components: {
        DropdownPanel,
    },

    setup() {
        return {
            args,
        };
    },

});

export const Basic = Component.bind({});
```

**Screenshot showing no controls**:
storybook

Am I missing something completely obvious?

### PR DESCRIPTION
**Pull request by andrewcourtice**
Fixes #46 

Added docgen info to Vite (via a custom plugin) output to automatically generate controls and docs for SFC components

**Note**: _This is a really basic implementation and will likely require further refinement._

### CODE REVIEW

File: packages/example-vue/stories/Button.stories.js [hunk 1]
```diff
@@ -3,13 +3,13 @@ import MyButton from './Button.vue';
...
     title: 'Example/Button',
     component: MyButton,
-    argTypes: {
-        backgroundColor: { control: 'color' },
-        size: {
-            control: { type: 'select', options: ['small', 'medium', 'large'] },
-        },
-        onClick: {},
-    },
+    // argTypes: {
```
Comment (eirslett) [inline packages/example-vue/stories/Button.stories.js:6]: Why is this code commented out?

Comment (andrewcourtice) [reply packages/example-vue/stories/Button.stories.js:6]: @eirslett - Apologies, I commented those out while testing to ensure the docgen info was generating the correct controls without the `argTypes` overriding it.

I'll fix it up now.

**Comment by IanVS** [comment]
@andrewcourtice thanks for the PR.  It looks like there are merge conflicts.  I think it's okay to add `vue-docgen-api` to the `dependencies` list, since it's not going to cause conflicts with storybook frameworks like what https://github.com/eirslett/storybook-builder-vite/pull/42 was intended to fix.

Long term, I wonder if we're going to need to come up with a better, pluggable architecture to support various frameworks. 🤔

**Comment by shilman** [comment]
@IanVS We're working on a metadata server in 6.4 that should handle this kind of stuff independently from webpack/vite etc. so i expect this problem to go away soon.

**Comment by eirslett** [comment]
I fixed the conflict and merged this manually!

**Pull request closed without merging**